### PR TITLE
abstracted the hooks

### PR DESCRIPTION
**Pull request by beefmaster**
Moved some logic around

### CODE REVIEW

File: src/hooks/carioHooks.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+
+export async function useMyNFTS(user: string) : Promise<NFTData[]> {
+  const nfts = await getNfts(user);
+
+  let localCache = bidsCache;
+
+  if (localCache.length === 0) {
+    localCache = await getAllBids();
+    bidsCache.push(...await getAllBids()); // update cache
```
Comment (michaelxuwu) [inline src/hooks/carioHooks.ts]: Can this just be `bidsCache.push(...localCache)` instead of a call again?

File: src/hooks/carioHooks.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  }
+
+  const usersLongs: PutData[] = [];
+
+  let myAddress = new BN(getStarknet().account.address.replace(/^0x/, ''), 16);
+  for(const each of localCache) {
+    if (each.seller_address === myAddress.toString(16)) {
```
Comment (michaelxuwu) [inline src/hooks/carioHooks.ts]: Should this be `.buyer_address`?

File: src/hooks/carioHooks.ts (hunk 1 above)
Comment (beefmaster) [reply src/hooks/carioHooks.ts]: yep good spot

**Comment by FawadHa1der** [comment]
Thanks @beefmaster

**Pull request merged**